squashfs-mount.c

```diff
@@ -77,17 +79,28 @@ int main(int argc, char **argv) {
 
   // Check that the mount point exists.
   int mnt_status = stat(mountpoint, &mnt_stat);
+
   if (mnt_status)
-    err(EXIT_FAILURE, "Invalid mount point \"%s\"", mountpoint);
+    exit_with_error("Invalid mount point \"%s\"", mountpoint);
+
   if (!S_ISDIR(mnt_stat.st_mode))
-    errx(EXIT_FAILURE, "Invalid mount point \"%s\" is not a directory", mountpoint);
+    exit_with_error("Invalid mount point \"%s\" is not a directory",
+                    mountpoint);
 
   // Check that the input squashfs file exists.
-  int sqsh_status = stat(squashfs_file, &mnt_stat);
+  int squashfs_fd = open(squashfs_file, O_RDONLY);
+
+  if (squashfs_fd < 0)
+    exit_with_error("Could not open squashfs file \"%s\"\n", squashfs_file);
+
+  int sqsh_status = fstat(squashfs_fd, &mnt_stat);
+
   if (sqsh_status)
-    err(EXIT_FAILURE, "Invalid squashfs image file \"%s\"", squashfs_file);
+    exit_with_error("Invalid squashfs image file \"%s\"\n", squashfs_file);
```

Comment:
The benefit of `err` in this context was that it gave a more detailed error message (e.g. that the path did not exist).

Reply:
Ah, actually I wasn't aware of err. It's a Linux kernel thing? We could use it since the code isn't portable anyways.

Reply:
https://linux.die.net/man/3/errx

`These functions are nonstandard BSD extensions.` that work on the few flavors of Linux that we have tried.